lib/trace-agent.js

```diff
@@ -247,25 +247,22 @@ TraceAgent.prototype.parseContextFromHeader = function(str) {
 };
 
 /**
- * Adds the provided trace context to the provided http headers
- * so it can follow the associated request through other
- * Google services.
+ * Generates a trace context header value that can be used
+ * to follow the associated request through other Google services.
  *
  * @param {SpanData} spanData The span to be added to headers.
- * @param {Object} headers The http headers associated with the
- *   current request.
  * @param {boolean} traced Whether this request was traced by the agent.
  */
-TraceAgent.prototype.addContextToHeaders = function(spanData, headers, traced) {
+TraceAgent.prototype.generateTraceContext = function(spanData, traced) {
   if (spanData === SpanData.nullSpan) {
-    return;
+    return '';
```

Comment:
We currently validate that we are tracing the request (`spanData !== SpanData.nullSpan`) before calling this method in all cases. The intent would be to check for empty string at any call site that cannot verify that `spanData` is not the null span before adding it to headers.